File: git/demo/backend/src/main/java/streamer_website/demo/entity/discord/JoinToCreateConfig.java
package streamer_website.demo.entity.discord;

import com.fasterxml.jackson.annotation.JsonProperty;
import discord4j.common.util.Snowflake;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Table(name = "discord_join_to_create_channels")
@Data
public class JoinToCreateConfig {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Trigger Channel
    @Column(name = "join_channel_id", nullable = false)
    private Long joinChannelId;

    @Column(name = "category_id", nullable = false)
    private Long categoryId;

    @Column(name = "channel_name_prefix", nullable = false)
    private String channelNamePrefix;

    @Column(name = "user_limit")
    private Integer userLimit;

    @Column(name = "is_private")
    @JsonProperty("privateChannel")
    private Boolean privateChannel;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    /** Hilfsmethode zum Konvertieren in Snowflake, falls du Discord4J-Objekte brauchst */
    public Snowflake joinChannelSnowflake() {
        return Snowflake.of(joinChannelId);
    }

    public Snowflake categorySnowflake() {
        return Snowflake.of(categoryId);
    }
}

File: git/demo/backend-go/internal/domain/join_to_create_config.go
package domain

import "time"

type JoinToCreateConfig struct {
	ID                int64     `json:"id" db:"id"`
	GuildID           int64     `json:"guildId,string" db:"guild_id"`
	UserID            string    `json:"userId" db:"user_id"`
	JoinChannelID     int64     `json:"joinChannelId,string" db:"join_channel_id"`
	CategoryID        int64     `json:"categoryId,string" db:"category_id"`
	ChannelNamePrefix string    `json:"channelNamePrefix" db:"channel_name_prefix"`
	UserLimit         int       `json:"userLimit" db:"user_limit"`
	PrivateChannel    bool      `json:"privateChannel" db:"private_channel"`
	Enabled           bool      `json:"enabled" db:"enabled"`
	CreatedAt         time.Time `json:"createdAt" db:"created_at"`
	UpdatedAt         time.Time `json:"updatedAt" db:"updated_at"`
}

type JoinToCreateConfigCreateInput struct {
	GuildID           int64  `json:"guildId,string"`
	UserID            string `json:"userId"`
	JoinChannelID     int64  `json:"joinChannelId,string"`
	CategoryID        int64  `json:"categoryId,string"`
	ChannelNamePrefix string `json:"channelNamePrefix"`
	UserLimit         int    `json:"userLimit"`
	PrivateChannel    bool   `json:"privateChannel"`
	Enabled           bool   `json:"enabled"`
}

type JoinToCreateConfigUpdateInput struct {
	JoinChannelID     *int64  `json:"joinChannelId,string"`
	CategoryID        *int64  `json:"categoryId,string"`
	ChannelNamePrefix *string `json:"channelNamePrefix"`
	UserLimit         *int    `json:"userLimit"`
	PrivateChannel    *bool   `json:"privateChannel"`
	Enabled           *bool   `json:"enabled"`
}